docs/docs/tutorial/creating-a-source-plugin/part-4/index.mdx

```diff
@@ -0,0 +1,865 @@
+    }
+    ```
+
+1. You can populate the error map like so:
+
+    - The object's keys are the unique IDs
+    - The object should contain these three keys and values:
+      - `text`: Your text that will be shown to the user for this specific error
+      - `level`: Set this to `ERROR`
+      - `category`: Set this to `THIRD_PARTY`
+
+    Add the following to `plugin/src/on-plugin-init.ts`:
+
+    ```ts:title=plugin/src/on-plugin-init.ts
+    import type { GatsbyNode } from "gatsby"
+    import { ERROR_CODES } from "./constants"
+
+    export const onPluginInit: GatsbyNode[`onPluginInit`] = ({ reporter }) => {
+      reporter.setErrorMap({
+        [ERROR_CODES.GraphQLSourcing]: {
+          text: (context) => `${context.sourceMessage}: ${context.graphqlError}`,
+          level: `ERROR`,
+          category: `THIRD_PARTY`,
+        },
+      })
+    }
+    ```
+
+    `text` has to be a function that returns a `string` and receives a freeform `context`. You can pass through any information you'd like in `context` and then use it to construct the final message that is shown to the user. So the error above requires `sourceMessage` and `graphqlError` to work.
+
+    <Announcement>
+
+    **Pro tip:** If your `text` should be a multi-line string, consider using the `stripIndent` function from [common-tags](https://www.npmjs.com/package/common-tags#stripindent). This way your error will be displayed correctly independent from how you write the string.
+
+    </Announcement>
+
+1. In this step you'll invoke the error with the custom ID and pass along the required `context` information. Open `plugin/src/source-nodes.ts` and add the `ERROR_CODES` import and use it inside the existing `panicOnBuild` call:
+
+    ```ts:title=plugin/src/source-nodes.ts
+    import type { GatsbyNode, SourceNodesArgs, NodeInput } from "gatsby"
+    import { fetchGraphQL } from "./utils"
+    import type { IAuthorInput, IPostInput, NodeBuilderInput } from "./types"
+    // highlight-next-line
+    import { NODE_TYPES, ERROR_CODES } from "./constants"
+
+    export const sourceNodes: GatsbyNode[`sourceNodes`] = async (gatsbyApi) => {
+      // Rest of sourceNodes
+
+      if (errors) {
+        // highlight-start
+        sourcingTimer.panicOnBuild({
+          id: ERROR_CODES.GraphQLSourcing,
+          context: {
+            sourceMessage: `Sourcing from the GraphQL API failed`,
+            graphqlError: errors[0].message,
+          },
+        })
+        // highlight-end
+      }
+
+      // Rest of sourceNodes
+    }
+
+    // Rest of file
+    ```
+
+1. As in step 3 of [Task: Output errors](#task-output-errors), add a typo to the GraphQL error again (e.g. `title2`), restart the `develop:site` script and you should see a new output for the error:
+
+    ```shell
+    ERROR #plugin_10000  PLUGIN
+
+    Sourcing from the GraphQL API failed: Cannot query field "title2" on type "Post". Did you mean "title"?
+
+    failed Sourcing from plugin API - 0.029s
+    ```
```

Comment:
I'm pretty confident I have all the code changes applied that are covered in this section, however the error message I see in my terminal is:
<img width="784" alt="Screen Shot 2023-03-20 at 6 16 14 PM" src="https://user-images.githubusercontent.com/1790506/226477605-c99947cd-1c2a-4e23-97ad-e41c1a2f7008.png">

I just wanted to call out the discrepancy — might not be of great concern.

Reply:
Oh, then things are not working as expected and we should dig into this!

You need to see `ERROR #plugin_10000  PLUGIN` otherwise the custom error is not working correctly.

Reply:
Okay I finally got around to revisiting this. I went through Part 4 again and double checked the code and I'm still not getting the custom error.

I pushed up what I'm working with locally here: https://github.com/aghreed/source-plugin-tutorial/tree/start